### PR TITLE
fix: no-dupe-keys with legal duplicated keys (fixes #133)

### LINKED ISSUE #133: The no-dupe-keys rule complains about legal duplicated keys in a declared class
Take, for instance, this Flow class:

``` Javascript
//@flow
...
declare class ProjectClass extends Map<any, any> {
  get(key: 'id'): number;
  get(key: 'children'): List<Project>;
}
```

The ESLint engine complains, 

```
flowtype/no-dupe-keys Duplicate property.: /Users/pairing/[...]/ProjectModalBody.js:1
```

However, it's allowed to have two `get` methods, as they are differentiated with different `keys`.  It would be great for the dupe rule to be aware of this and allow it.

### PR DESCRIPTION
**Pull request by object88**
Addresses #133 by validating a complex class structure with multiple methods and different keys.

### CODE REVIEW

**Comment by object88** [comment]
Bump; didn't see that merge errors were introduced.  Have been fixed up now.

**Comment by object88** [comment]
Updated my fork to keep up with master.  Can this be merged in?

File: src/rules/noDupeKeys.js [hunk 1]
```diff
@@ -12,13 +12,68 @@ export default (context) => {
...
+  const analizeElement = (element) => {
+    const {type} = element;
+    let value;
+
+    switch (type) {
+    case 'GenericTypeAnnotation':
+      value = element.id.name;
+      break;
+    case 'ObjectTypeAnnotation':
+      /* eslint-disable no-use-before-define */
```
Comment (gajus) [inline src/rules/noDupeKeys.js]: Please change `/* eslint-disable */` comments to `// eslint-disable-next-line no-use-before-define`.

**Comment by object88** [comment]
Change made; thanks for the feedback.

**Comment by object88** [comment]
Are there any other changes that I should make?

**Comment by object88** [comment]
I seem to have done something wrong on my end, in my fork.  Will resolve this.

**Comment by gajus** [comment]
@object88 Ping me when it is ready to be re-reviewed.

**Comment by object88** [comment]
OK, we are back in working order.  Feel free to check it out, @gajus , and thanks!

**Comment by gajus** [comment]
Thank you

**Pull request merged**